documentation/Architecture/Overview.md

```diff
@@ -2,80 +2,105 @@
 
 This document outlines the architecture of the Microsoft Performance Toolkit SDK.
 
-For more detailed information on how to create your own project using the SDK, please view [Creating your own project](../Using-the-SDK/Creating-your-project.md). 
+For more detailed information on how to create your own project using the SDK, please view [Creating an SDK Plugin C# Project](../Using-the-SDK/Creating-your-project.md). 
 
-Examples:
-- [Custom Data Source Example](../../samples/SimpleDataSource/SimpleCustomDataSource.cs).
-- [Custom Data Processor Example](../../samples/SimpleDataSource/SimpleCustomDataProcessor.cs). 
 
+## High-Level Interfaces
 
-# Microsoft Performance Toolkit SDK
+The Microsoft Performance Toolkit SDK was built to empower users to analyze arbitrary data source (such as `.etl` files, `.ctf` files, SQL servers, etc). 
+Users can develop *SDK plugins* that contain logic for processing data sources into tables, which are sent to the *SDK driver* (such as 
+the [Windows Performance Analyzer](https://docs.microsoft.com/en-us/windows-hardware/test/wpt/windows-performance-analyzer)) to render. At the highest 
+level, it exposes the following interfaces:
 
-The Microsoft Performance Toolkit SDK was built to empower users to analyze any arbitrary data source (e.g. .etl, .ctf, etc). 
-Using the SDK, any data source (including large files) can be quickly processed to generate custom graphs and tables.
-Below is a brief overview of the structure of the SDK.
+<img src=".attachments/overview.svg" width="800">
 
 
-![](.attachments/ArchitectureOverview.png)
+The interfaces depicted in this diagram are
 
+1) The SDK takes as input *data sources* (such as file paths) from the *SDK driver*. This input is typically generated from the user, such as when the user asks 
+the driver to open a file.
```

Comment:
The sentence is a bit hard to me. "from the user" seems unimportant. If the input is the data read from the file, just saying SDK driver converts the source to data would be sufficient?

Reply:
The input is actually the path to the file, not its content. I will reword the second sentence since I agree it reads kind of awkwardly as-is :)